myapp/main.py

```diff
@@ -0,0 +1,474 @@
+    for lab in query_all_lab:
+        lab_list.append({'lab_name':lab.lab_name,'class_name':lab.class_name,'prof_name':lab.prof_name})   
+    return render_template('student_enter_data.html',lab_list=lab_list)    
+
+# A student can input data here
+@app.route('/student_data_entry/<lab_id>')
+def student_data_entry(lab_id):
+    rows_info = []
+    lab_info = []
+    db_session = db.get_session()
+    query_lab_info = db_session.query(schema.Lab_info).filter(schema.Lab_info.lab_id==lab_id)
+    query_rows_info = db_session.query(schema.Lab_rows).filter(schema.Lab_rows.lab_id==lab_id).order_by(schema.Lab_rows.row_order)
+
+    #Convert the objects to dictionaries to make them JSON serializable
+    for row in query_rows_info:
+        rows_info.append({'row_name':row.row_name,'row_desc':row.row_desc,'value_type':row.value_type,'value_range':row.value_range,'value_candidates':row.value_candidates})
+    for lab in query_lab_info:
+        lab_info.append({'lab_id':lab.lab_id,'lab_name':lab.lab_name,'lab_desc':lab.lab_desc})
+    return render_template('student_data_entry.html',rows_info=rows_info,lab_info=lab_info)    
+
+
+# store incoming data into the database
+@app.route('/_student_receive_data', methods=['POST'])
+def _student_receive_data():
+    jsonData = request.get_json()
+    lab_data = jsonData['lab_data']
+    db_session = db.get_session()
+    for d in lab_data:
+        student_name = d['student_name']     
+        for r in d['rows_info']:
+            lab_id = r['lab_id']
+            row_name = r['row_name']
+            row_id = Generate_row_id(lab_id,row_name)
+            data_id = Generate_data_id(row_id,student_name)
+            db_session.add(schema.Lab_data(row_id=row_id,data_id=data_id,student_name=student_name,row_data=r['row_data']))
+    db_session.commit()
+    # return success to the ajax call from flask 
+    return jsonify(success=True, data=jsonData)
+
+
+
+
+
+# 4.Admin Side
+# admin homepage
+@app.route('/admin_home')
+def admin_home():
+    return render_template('admin_home.html')
+
+# create a new lab/manage current labs
+@app.route('/admin_setup_labs_and_data_access', methods=['GET', 'POST'])
+def admin_setup_labs_and_data_access():
+    # Collect Info of all the existing labs
+    current_labs = {'activated':[],'downloaded':[],'unactivated':[]}
+    current_labs['activated']
+    current_labs['downloaded']
+    current_labs['unactivated']
+
+    db_session = db.get_session()
+    query_activated = db_session.query(schema.Lab_info).filter(schema.Lab_info.lab_status=='Activated')
+    query_downloaded = db_session.query(schema.Lab_info).filter(schema.Lab_info.lab_status=='Downloaded')
+    query_unactivated = db_session.query(schema.Lab_info).filter(schema.Lab_info.lab_status=='Unactivated')
+    
+     
+    for lab_a in query_activated:
+        current_labs['activated'].append({'lab_id':lab_a.lab_id,'lab_name':lab_a.lab_name,'class_name':lab_a.class_name,'prof_name':lab_a.prof_name})
```

Comment:
It's a good idea to write methods that serialize and deserialize from your database model objects to JSON. Then, if your schema has to change you only have to change the method to update the context passed to all the views.